feature_engine/sanity_check/similar_columns.py

```diff
@@ -0,0 +1,68 @@
+from typing import Any, List
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class SimilarColumns(BaseEstimator, TransformerMixin):
+    """Ensure that similar columns are in test and train dataset.
+
+
+    Parameters
+    ----
+    None
+
+    """
+
+    def __init__(self, impute_with: Any = np.nan):
+        self.impute_with = impute_with
+        self.col: List[str] = []
+
```

Comment:
We don't need to define this attribute in the init, not per the sklearn / feature_engine conventions